source-oracle/backfill.go

```diff
@@ -240,21 +381,34 @@ func castColumn(col sqlcapture.ColumnInfo) string {
 // Keyless scan uses ROWID to order the rows. Note that this only ensures eventual consistency
 // since ROWIDs are not always increasing (new rows can use smaller ROWIDs if space is available in an earlier block)
 // but since we will capture changes since the start of the backfill using SCN tracking, we will eventually be consistent
-func (db *oracleDatabase) keylessScanQuery(info *sqlcapture.DiscoveryInfo, schemaName, tableName string) string {
+func (db *oracleDatabase) keylessScanQuery(info *sqlcapture.DiscoveryInfo, schemaName, tableName, afterRowID string) (string, string) {
 	var query = new(strings.Builder)
 	var columnSelect []string
 	for _, col := range info.Columns {
 		columnSelect = append(columnSelect, castColumn(col))
 	}
 
+	var streamID = sqlcapture.JoinStreamID(schemaName, tableName)
+
+	var idx = slices.Index(db.backfillRowIDRanges[streamID], afterRowID)
```

Comment:
nit: I don't think there's any actual problem in practice, but it might be cleaner to pass in `db.backfillRowIDRanges[streamID]` as an argument to the `keylessScanQuery` function, just to make the data/control flow clearer.